p2p/node/streamManager/streamManager.go

```diff
@@ -146,34 +149,40 @@ func (sm *basicStreamManager) listenForNewStreamRequest() {
 }
 
 func (sm *basicStreamManager) OpenStream(peerID p2p.PeerID) error {
-	// check if there is an existing stream
+	// Check if there is an existing stream
 	if _, ok := sm.streamCache.Get(peerID); ok {
 		return nil
 	}
-	timer := time.NewTimer(c_stream_timeout)
-	defer timer.Stop()
-	select {
-	case <-timer.C:
-		return errors.New("stream creation timeout")
-	default:
-		// Create a new stream to the peer and register it in the cache
-		stream, err := sm.host.NewStream(sm.ctx, peerID, quaiprotocol.ProtocolVersion)
-		if err != nil {
-			return fmt.Errorf("error opening new stream with peer %s err: %s", peerID, err)
-		}
-		wrappedStream := streamWrapper{
-			stream:    stream,
-			semaphore: make(chan struct{}, c_maxPendingRequests),
-			errCount:  0,
-		}
-		sm.streamCache.Add(peerID, wrappedStream)
-		go quaiprotocol.QuaiProtocolHandler(stream, sm.p2pBackend)
-		log.Global.WithField("PeerID", peerID).Info("Had to create new stream")
-		if streamMetrics != nil {
-			streamMetrics.WithLabelValues("NumStreams").Inc()
+
+	streamCtx, streamCancel := context.WithTimeout(sm.ctx, c_stream_timeout)
```

Comment:
i dont think this is useful

Reply:
I removed this but now realize its actually necessary